src/test/java/com/codeus/winter/config/DefaultBeanFactoryTest.java

```diff
@@ -628,6 +628,22 @@ void testFailInitializeBeanWithPrototypeScope() {
 
     }
 
+    @Test
+    @DisplayName("Should initialize a bean with multiple candidates using @Qualifier annotation")
+    void testShouldInitializeBeanWithMultipleCandidatesUsingQualifier() {
+        BeanDefinition beanDefinitionWithQualifierAnnotation = new BeanDefinitionImpl();
+        beanDefinitionWithQualifierAnnotation.setBeanClassName("com.codeus.winter.test.BeanWithQualifierAnnotation");
```

Comment:
Please use mock so the code follows the same format. You can use the `beanDefinitionMock` util method to create a bean definition mock for singleton bean.

Reply:
fixed

Reply:
Line with `beanDefinitionWithQualifierAnnotation.setBeanClassName(...)` can be also removed because BeanClassName is set within the `singletonBeanDefinitionMock` method.

Reply:
Still relevant, now it is line 717.

Reply:
fixed